src/server/client.ts

```diff
@@ -719,20 +719,42 @@ namespace ts.server {
             return response.body;
         }
 
-        getRefactorCodeActions(
+        getEditsForRefactor(
             fileName: string,
             _formatOptions: FormatCodeSettings,
             positionOrRange: number | TextRange,
-            refactorName: string) {
+            refactorName: string,
+            actionName: string): RefactorEditInfo {
 
-            const args = this.createFileLocationOrRangeRequestArgs(positionOrRange, fileName) as protocol.GetRefactorCodeActionsRequestArgs;
-            args.refactorName = refactorName;
+            const args = this.createFileLocationOrRangeRequestArgs(positionOrRange, fileName) as protocol.GetEditsForRefactorRequestArgs;
+            args.refactor = refactorName;
+            args.action = actionName;
 
-            const request = this.processRequest<protocol.GetRefactorCodeActionsRequest>(CommandNames.GetRefactorCodeActions, args);
-            const response = this.processResponse<protocol.GetRefactorCodeActionsResponse>(request);
-            const codeActions = response.body.actions;
+            const request = this.processRequest<protocol.GetEditsForRefactorRequest>(CommandNames.GetEditsForRefactor, args);
+            const response = this.processResponse<protocol.GetEditsForRefactorResponse>(request);
+            const edits: FileTextChanges[] = this.convertCodeEditsToTextChanges(response.body.edits);
 
-            return map(codeActions, codeAction => this.convertCodeActions(codeAction, fileName));
+            const renameFilename: string | undefined = response.body.renameFilename;
+            let renameLocation: number | undefined = undefined;
+            if (renameFilename !== undefined) {
+                renameLocation = this.lineOffsetToPosition(renameFilename, response.body.renameLocation);
+            }
+
+            return {
+                edits,
+                renameFilename,
+                renameLocation
+            };
+        }
+
+        private convertCodeEditsToTextChanges(edits: ts.server.protocol.FileCodeEdits[]): FileTextChanges[] {
+            return edits.map(edit => {
+                const fileName = edit.fileName;
+                return {
+                    fileName,

```

Comment:
The end result here is an array that looks like this (field names not exact):
```
[
  { filename: "foo.ts",
    changes: [ { pos: 10, len: 5, newText "bar" },
                     { pos: 17, len: 2, newText "baz" }
  },
  { filename: "bar.ts",
    changes: [ { pos: 20, len: 2, newText "qar" },
                     { pos: 27, len: 6, newText "qaz" }
  }
]
```
`convertTextChangeToCodeEdit` needs `filename` to do pos <-> line/col mapping, not to put it in the resulting object